tlspuffin/src/integration_tests/determinism.rs

```diff
@@ -0,0 +1,58 @@
+use puffin::{put::PutOptions, trace::TraceContext};
+
+use crate::{
+    put_registry::TLS_PUT_REGISTRY,
+    tls::{seeds::seed_client_attacker_full, trace_helper::TraceHelper},
+};
+
+#[test]
+#[cfg(all(
+    feature = "deterministic",
+    feature = "boringssl-binding",
+    feature = "tls13",
+    feature = "TODO"
+))] // TODO: only passes in mono-thread!! with option `-test-threads=1`
```

Comment:
Should we ignore this test?

Reply:
We'd like to keep this test to be run manually in mono-thread mode. We cannot include it non-conditionally because the CI will fail otherwise as it is not in mono thread. Adding a mono-thread feature does not seem ideal either.